### ISSUE TITLE
Payment only works once then hangs

### ISSUE THREAD

**Issue by rivella50** [PurpleSoftSrl/sumup_flutter_plugin#36]
Hi there,
i've just updated to version 0.6.0 and have the problem that a SumUp payment only once works fine. When trying the 2nd time it just tries to connect to the card reader and never finishes - although the bluetooth sign appears on the card reader's display.
Only when i shutdown the app it works once again.
The problem only seems to happen on Android.
This is the output i get:
```
D/BluetoothAdapter(26346): isLeEnabled(): ON
D/BluetoothGatt(26346): connect() - device: F4:94:E3:9C:3B:32, auto: false
D/BluetoothGatt(26346): registerApp()
D/BluetoothGatt(26346): registerApp() - UUID=1941db0a-0c8c-4d08-88ec-b4c6e3cb6df6
D/BluetoothGatt(26346): onClientRegistered() - status=0 clientIf=6
D/BluetoothGatt(26346): onClientConnectionState() - status=0 clientIf=6 device=F4:94:E3:9C:3B:32
D/BluetoothGatt(26346): requestConnectionPriority() - params: 1
D/BluetoothGatt(26346): discoverServices() - device: F4:94:E3:9C:3B:32
D/BluetoothGatt(26346): onConnectionUpdated() - Device=F4:94:E3:9C:3B:32 interval=12 latency=0 timeout=500 status=0
D/BluetoothGatt(26346): onConnectionUpdated() - Device=F4:94:E3:9C:3B:32 interval=6 latency=0 timeout=500 status=0
D/BluetoothGatt(26346): onSearchComplete() = Device=F4:94:E3:9C:3B:32 Status=0
D/BluetoothGatt(26346): setCharacteristicNotification() - uuid: 22ffc547-1bef-48e2-aa87-b87e23ac0bbd enable: true
D/BluetoothGatt(26346): onConnectionUpdated() - Device=F4:94:E3:9C:3B:32 interval=12 latency=0 timeout=500 status=0
D/BluetoothGatt(26346): onConnectionUpdated() - Device=F4:94:E3:9C:3B:32 interval=48 latency=0 timeout=800 status=0
D/BluetoothGatt(26346): setCharacteristicNotification() - uuid: 1f6b14c9-97fa-4f1e-aaa6-7e152fdd04f4 enable: true
D/BluetoothGatt(26346): cancelOpen() - device: F4:94:E3:9C:3B:32
D/BluetoothGatt(26346): onClientConnectionState() - status=0 clientIf=6 device=F4:94:E3:9C:3B:32
D/BluetoothGatt(26346): close()
D/BluetoothGatt(26346): unregisterApp() - mClientIf=6
```
So actually no real error.
Could you please try on your side if this is a general problem?

I've tested with an app version, where i still use version 0.5.0 and there several payments just work fine.

**Comment by sstasi95** [comment]
Hi,

Can you try with the example app? I've just tried many payments and it works fine.

**Comment by rivella50** [comment]
With the example app it seems to work. Very strange.
So it seems i have to investigate further on my side and check where the problem is - perhaps it has to do with another package i use (e.g. `flutter_local_notifications`).

**Comment by rivella50** [comment]
@sstasi95 Ok i guess that i've found the issue: Calling init again after a successful payment does lead to the hanging effect.
These are the steps how you should be able to reproduce it in the example app as well:
- click init
- click login
- click checkout and perform a payment
- after returning click init again
- click checkout -> now the process should stop at 'Verify Card Reader...' without showing any error

Imo the problem is that you do not check the private variable `_isInitialized` for a true value in method `init()`.
And since there is also no public getter for me to check if that variable is already true the problem arises, since i always call init before performing a payment in my app - which actually worked fine before version 0.6.0.

I hope that you can reproduce the problem on your side and offer a fix.

**Comment by sstasi95** [comment]
Fixed in 0.6.1

**Comment by rivella50** [comment]
That's great. Thank you very much.
